# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webcodecs/AudioData.h
# Commit: 94c203600f2c
# Full Hash: 94c203600f2c7090fb8d279bd12417f4f2f8d7af
# Author: Paul Adenot <paul@paul.cx>
# Date: 2024-03-06 21:16:27
# Regressor Bug: 1858958
# File Overlap Count: 1
# Description:
#   Bug 1858958 - AudioData: structured cloning. r=chunmin,smaug
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D190948
# ==============================================================================

diff -r 8f0d8d45d132 -r 94c203600f2c dom/media/webcodecs/AudioData.h
--- a/dom/media/webcodecs/AudioData.h	Wed Mar 06 11:14:17 2024 +0000
+++ b/dom/media/webcodecs/AudioData.h	Wed Mar 06 11:14:17 2024 +0000
@@ -7,16 +7,12 @@
 #ifndef mozilla_dom_AudioData_h
 #define mozilla_dom_AudioData_h
 
-#include "AudioSampleFormat.h"
 #include "WebCodecsUtils.h"
 #include "js/TypeDecls.h"
-#include "mozilla/Attributes.h"
 #include "mozilla/ErrorResult.h"
-#include "mozilla/NotNull.h"
 #include "mozilla/Span.h"
 #include "mozilla/dom/AudioDataBinding.h"
 #include "mozilla/dom/BindingDeclarations.h"
-#include "mozilla/dom/TypedArray.h"
 #include "mozilla/dom/StructuredCloneHolder.h"
 #include "nsCycleCollectionParticipant.h"
 #include "nsWrapperCache.h"
@@ -128,7 +124,7 @@
 
 class AudioDataResource final {
   NS_INLINE_DECL_THREADSAFE_REFCOUNTING(AudioDataResource);
-  AudioDataResource(FallibleTArray<uint8_t>&& aData)
+  explicit AudioDataResource(FallibleTArray<uint8_t>&& aData)
       : mData(std::move(aData)) {}
 
   explicit AudioDataResource()
@@ -155,7 +151,7 @@
 };
 
 struct AudioDataSerializedData {
-  AudioDataSerializedData(const AudioData& aFrom)
+  explicit AudioDataSerializedData(const AudioData& aFrom)
       : mTimestamp(aFrom.Timestamp()),
         mNumberOfChannels(aFrom.NumberOfChannels()),
         mNumberOfFrames(aFrom.NumberOfFrames()),
